llama_index/indices/service_context.py

```diff
@@ -111,6 +113,21 @@ def from_defaults(
             )
             chunk_size = chunk_size_limit
 
+        if isinstance(embed_model, str):
+            if embed_model != "local":
+                raise ValueError(
+                    "embed_model must be str 'local' or of type BaseEmbedding"
+                )
+            try:
+                from langchain.embeddings import HuggingFaceEmbeddings
+            except ImportError as exc:
+                raise ImportError(
+                    "Could not import sentence_transformers or langchain package. "
+                    "Please install with `pip install sentence_transformers langchain`."
```

Comment:
I'm assuming sentence_transformers also includes the transformers package? I think torch might also be a dependency here

Reply:
Hmm true. Turns out it's quite a long list :cry: 

`Requires: huggingface-hub, nltk, numpy, scikit-learn, scipy, sentencepiece, torch, torchvision, tqdm, transformers`

Reply:
I think most of these are cross depending 🤔 there should be a small-ish subset that ends up installing everything you need

Reply:
I think these are the only 2